src/nsupdate/main/templates/main/includes/tabbed_router_configuration.html

```diff
@@ -247,6 +248,96 @@ <h5>{% trans "Enter the following data (and click ADD afterwards):" %}</h5>
+## IPv4 Update code begin.
+:local DDNSHostnameIPv4 [:resolve $DDNSHostname];
+{
+:local IPV4CURRENT [/tool fetch url="https://ipv4.nsupdate.info/myip" as-value output=user];
+ :delay 2s; 
+ 
+ :put "IPv4 Hostname: $DDNSHostname " ;
+ :put "DNS-IPv4: $DDNSHostnameIPv4 " ;
+ :put "Current IPv4 address:" ;
+ :put ($IPV4CURRENT->"data") ;
+ # Compare DNS IPv4 to current IPv4
+ :if ($IPV4CURRENT->"data" = $DDNSHostnameIPv4 ) do={
+               :log info "IPv4 address matches for $DDNSHostname no update needed." ;
+               :put "IPv4 address matches for $DDNSHostname no update needed." ;
+        } else={
+            :log info "IPv4 address doesn't match for $DDNSHostname , updating..." ;
+            :put "IPv4 address doesn't match for $DDNSHostname , updating..." ;
+            
+            /tool fetch url="https://$DDNSHostname:$DDNSSecret@ipv4.nsupdate.info/nic/update" keep-result=no ;
+        }
+}
+## IPv4 Update code end
+:put "-----------------------------------------"
+# IPv6 Update code start, IPv6 code is much more complicated vs IPv4 code.
+# This is due to the around Mikrotik not implementing ping6 or resolve6 functionality
+
+# Setup some variables to generate a unique-ish address-list name
+:local AddressListName
+:local date [/system clock get date]
+:local newdate "";
+  :for i from=0 to=([:len $date]-1) do={ :local tmp [:pick $date $i];
+  :if ($tmp !="/") do={ :set newdate "$newdate$tmp" }
+  :if ($tmp ="/") do={}
+}
+
+:set AddressListName ($"DDNSHostname" . "-" . $"newdate" )
+
+# Create temporary address list to resolve IPv6 ip since Mikrotik resolve function sucks
+# Use uniqueish name for address list to prevent clobbering.
+/ipv6 firewall address-list add list=$AddressListName address=$DDNSHostname timeout=15s comment="Temp list for DDNS"
+:delay 2s;
+
+{
+# Get IPV6 string from address list.
+local DDNSHostnameIPv6 [/ipv6 firewall address-list get value-name=address [find comment=$DDNSHostname]] 
+# Split IPv6 (address) string away from its subnet/cidr mask. 
+local DDNSHostnameIPv6 [:pick $DDNSHostnameIPv6 0 [:find $DDNSHostnameIPv6 "/"]]
+# Get local IPv6 address from nsupdate.info server.
+:local IPV6CURRENT [/tool fetch url="https://ipv6.nsupdate.info/myip" as-value output=user];
+ :delay 2s;
+ :put "IPv6 Hostname: $DDNSHostname " ;
+ :put "DNS-IPv6: $DDNSHostnameIPv6 " ;
+ :put "Current IPv6 address:"
+ :put ($IPV6CURRENT->"data");
+# Compare DNS IPv6 to current IPv6
+:if ( $IPV6CURRENT->"data" = $DDNSHostnameIPv6 ) do={
+            :log info "IPv6 address matches for $DDNSHostname , no update needed." ;
+            :put "IPv6 address matches for $DDNSHostname, no update needed." ;
+        } else={
+            :log info "IPv6 address doesn't match for $DDNSHostname , updating..." ;
+            :put "IPv6 address doesn't match for $DDNSHostname, updating..." ;
+            /tool fetch url="https://$DDNSHostname:$DDNSSecret@ipv6.nsupdate.info/nic/update" keep-result=no ;
```

Comment:
you need to evaluate the http return code of this and switch local state as needed. please have a look at the dyndns "standard" about this.

Reply:
Haven't had time to work on this again, its on my todo.

Reply:
Did you fix this meanwhile?

I think it would be nice if we showed a config for microtik, but evaluation of the return code as defined by the standard is a requirement (needed so update client does not DoS the dyndns service in case something is going wrong / is misconfigured).